libsolidity/interface/CompilerStack.h

```diff
@@ -221,6 +225,13 @@ class CompilerStack: public langutil::CharStreamProvider
 	/// @returns false on error.
 	bool analyze();
 
+	/// Perform the analysis steps of legacy language mode.
+	/// @returns false on error.
+	bool analyzeLegacy(bool _noErrorsSoFar);
+	/// Perform the analysis steps of experimental language mode.
+	/// @returns false on error.
+	bool analyzeExperimental();
```

Comment:
I think these should be in the private section of the class, not exposed in the public API. Currently it's quite unclear how they relate to `analyze()` - should you use them instead? Afterwards? The docstring does not clarify that either.

They're just an implementation detail of `analyze()`, aren't they?

Reply:
True, making them private makes sense.

Reply:
Moved to private section of class.